app/src/main/java/cl/maleb/testcarsales/ui/covid/CovidFragment.kt

```diff
@@ -0,0 +1,112 @@
+package cl.maleb.testcarsales.ui.covid
+
+import android.app.DatePickerDialog
+import android.os.Bundle
+import android.view.LayoutInflater
+import androidx.lifecycle.Observer
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModelProvider
+import cl.maleb.testcarsales.data.Result
+import cl.maleb.testcarsales.databinding.CovidLayoutBinding
+import cl.maleb.testcarsales.di.Injectable
+import cl.maleb.testcarsales.di.injectViewModel
+import java.util.*
+import javax.inject.Inject
+
+class CovidFragment : Fragment(), Injectable {
+    @Inject
+    lateinit var viewModelFactory: ViewModelProvider.Factory
+
+    private lateinit var viewModel: CovidViewModel
+    private lateinit var binding: CovidLayoutBinding
+
+    // Calendar variables
+    val c = Calendar.getInstance()
```

Comment:
Isn't there a way to move this variables to another class? CovidFragment shouldn't know about calendar variables

Reply:
I think that I'll need to do DateUtils to return this data. Like functions based in calendar that returns day, month and year by separately. I need that because is required to form a datepicker dialog. What do you think ?

Reply:
You definitely need remove this from fragment, you can consider moving the dialog creation away from the fragment too, using a Kotlin extensions, for example, take a look at this page: https://android.jlelse.eu/android-custom-alert-dialogs-kotlin-extension-functions-kotlin-higher-order-functions-life-682305c5322e

Reply:
Indeed, I removed completely from the Fragment